sips/sip-012/scripts/extract-stackers.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+
+############################################################
+# Script to extract all STX addresses that are stacking
+# in a particular reward cycle, given access to a node's
+# debug log file.
+#
+# Usage: ./extract-stackers.sh REWARD-CYCLE < DEBUG-LOG-FILE
+############################################################
+
+set -oue pipefail
+
+FIRST_BLOCK=666050
+REWARD_CYCLE_LENGTH=2100
+
+# DEBG [1635829849.697968] [src/chainstate/stacks/db/accounts.rs:298] [chains-coordinator] PoX lock 604260000 uSTX (new balance 6104) until burnchain block height 689150 for Standard(StandardPrincipalData(SP2WGGG8E7NVFA0W4YZ87M6R09TNJ33CEBAXB9QPB))
+# DEBG [1635941010.356021] [src/chainstate/stacks/db/accounts.rs:298] [chains-coordinator] PoX lock 12622260119595 uSTX (new balance 0) until burnchain block height 714350 for Contract(QualifiedContractIdentifier { issuer: StandardPrincipalData(SP2C2YFP12AJZB4MABJBAJ55XECVS7E4PMMZ89YZR), name: ContractName("arkadiko-stacker-v1-1") })
+#
+# becomes
+#
+# {"type";"stacker", "locked":"604260000","until":"689150","address":"SP2WGGG8E7NVFA0W4YZ87M6R09TNJ33CEBAXB9QPB"}
+# {"type":"stacker", "locked":"12622260119595","until":"714350","address":"SP2C2YFP12AJZB4MABJBAJ55XECVS7E4PMMZ89YZR.arkadiko-stacker-v1-1"}
+extract_stacker() {
+   # stdin: a DEBUG-level logfile
+   # stdout: JSON describing each PoX lock event
+   gawk '
+{
+   num_locked = $7
+   until = $16
```

Comment:
The success of this function appears to be tied directly to the token quantity and ordering found inside debug log output, which looks brittle and a source of future bugs – if/when someone edits the upstream debug message, not realizing the downstream implications here. So, add some sanity checks before proceeding to use "num_locked" and "until" values?

Reply:
All of the stacking records for the vote have been extracted and stored as supplementary materials to this SIP, so there's not much of a concern here.  This script is meant mainly to explain how the votes were extracted at the time they were extracted.